subsys/net/ip/tcp2.c

```diff
@@ -1083,46 +1090,61 @@ static struct tcp *tcp_conn_alloc(void)
 
 	ret = k_mem_slab_alloc(&tcp_conns_slab, (void **)&conn, K_NO_WAIT);
 	if (ret) {
+		NET_ERR("Cannot allocate slab");
 		goto out;
 	}
 
 	memset(conn, 0, sizeof(*conn));
 
+	if (CONFIG_NET_TCP_RECV_QUEUE_TIMEOUT) {
```

Comment:
I see that this code is just moved around, but as you already add comments, can you add something here? For example, I see that we preallocate a pkt for send data, so other connections can never starve us up on that (assuming that we keep ownership of that packet to the connection). That's  :+1:, and worth being mentioned explicitly in the comment?

Reply:
The code was moved around so that if there is an allocation error, we notice it early and not do any other initializations done by that func. I see no reason for any extra comments here.